azure-pipelines/builds/ci.yml

```diff
@@ -15,24 +15,69 @@ variables:
   - name: Codeql.Enabled
     value: true
 
+  - name: defaultPoolName
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore-Svc-Public' ), False, 'NetCore-Public')]
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: $[replace(replace(eq(contains(coalesce(variables['System.PullRequest.TargetBranch'], variables['Build.SourceBranch'], 'refs/heads/main'), 'release'), 'true'), True, 'NetCore1ESPool-Svc-Internal'), False, 'NetCore1ESPool-Internal')]
+
+  - name: defaultPoolDemandsLinux
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: ImageOverride -equals Build.Ubuntu.1804.Amd64.Open
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: ImageOverride -equals Build.Ubuntu.1804.Amd64
+
+  - name: defaultPoolDemandsWindows
+    ${{ if eq(variables['System.TeamProject'], 'public') }}:
+      value: ImageOverride -equals 1es-windows-2022-open
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      value: ImageOverride -equals 1es-windows-2022
+
 stages:
 - stage: build
   displayName: Build
   jobs:
-  - template: /eng/common/templates/jobs/jobs.yml
-    parameters:
-      enablePublishUsingPipelines: true
-      enablePublishBuildArtifacts: true
-      enablePublishBuildAssets: true
-      publishAssetsImmediately: true
-      artifacts:
-        publish:
-          artifacts: true
-          manifests: true
-      enableSourceBuild: true
-  - template: /azure-pipelines/builds/generatescript-tests.yml
-    parameters:
-      buildConfiguration: Debug
+  - job: Linux
+    pool:
+      name: ${{ variables.defaultPoolName }}
+      demands: ${{ variables.defaultPoolDemandsLinux }}
+    container: mcr.microsoft.com/dotnet-buildtools/prereqs:centos-stream8
+    steps:
+    - template: /eng/common/templates/steps/source-build.yml
+
+    - script: $(Build.SourcesDirectory)/test.sh
+      displayName: Run Tests
```

Comment:
Interesting. I would expect the `source-build.yml` file to support also running tests. It might make sense to add such a feature as an opt-in.

Reply:
Same for the publish test results. The default `build` step supports running and publishing test results (and with different formats), i.e. here: https://github.com/dotnet/arcade/blob/69e73774ddfdacb1eea03e86d0c6ffcdb94eb4dd/eng/common/templates/job/job.yml#L32

cc @MichaelSimons @mthalman

Reply:
Would it make more sense to add non source build legs here that then also run the tests and keep the existing source build legs (which wouldn't run tests) separate?

Reply:
I don't see value in running a non source-build leg for this repo since it's intended purpose is solely for source-build.  I do agree it would make sense to have an option to support running tests in the source-build leg.

Reply:
> It might make sense to add such a feature as an opt-in.

@ViktorHofer I opened https://github.com/dotnet/arcade/issues/14220 which relates to this suggestion.